mantle/cmd/kola/testiso.go

```diff
@@ -679,55 +679,104 @@ func awaitCompletion(ctx context.Context, inst *platform.QemuInstance, outdir st
+			for _, exp := range expected {
+				l, err := r.ReadString('\n')
+				if err != nil {
+					if err == io.EOF {
+						// this may be from QEMU getting killed or exiting; wait a bit
+						// to give a chance for .Wait() above to feed the channel with a
+						// better error
+						time.Sleep(1 * time.Second)
+						errchan <- fmt.Errorf("Got EOF from completion channel, %s expected", exp)
+					} else {
+						errchan <- errors.Wrapf(err, "reading from completion channel")
+					}
+					return
 				}
-				return
+				line := strings.TrimSpace(l)
+				if line != exp {
+					errchan <- fmt.Errorf("Unexpected string from completion channel: %s expected: %s", line, exp)
+					return
+				}
+				plog.Debugf("Matched expected message %s", exp)
 			}
-			line := strings.TrimSpace(l)
-			if line != exp {
-				errchan <- fmt.Errorf("Unexpected string from completion channel: %s expected: %s", line, exp)
-				return
+			plog.Debugf("Matched all expected messages")
+			// OK!
+			errchan <- nil
+		}()
+		go func() {
+			//check for error when switching boot order
+			if booterrchan != nil {
+				if err := <-booterrchan; err != nil {
+					errchan <- err
+				}
 			}
-			plog.Debugf("Matched expected message %s", exp)
-		}
-		plog.Debugf("Matched all expected messages")
-		// OK!
-		errchan <- nil
-	}()
-	go func() {
-		//check for error when switching boot order
-		if booterrchan != nil {
-			if err := <-booterrchan; err != nil {
-				errchan <- err
+		}()
+		err := <-errchan
+		elapsed := time.Since(start)
+		if err == nil {
+			// No error so far, check the console and journal files
+			consoleFile := filepath.Join(outdir, "console.txt")
+			journalFile := filepath.Join(outdir, "journal.txt")
+			files := []string{consoleFile, journalFile}
+			for _, file := range files {
+				fileName := filepath.Base(file)
+				// Check if the file exists
+				_, err := os.Stat(file)
+				if os.IsNotExist(err) {
+					fmt.Printf("The file: %v does not exist\n", fileName)
+					continue
+				} else if err != nil {
+					fmt.Println(err)
+					errchan <- err
+					return elapsed, err
+				}
+				// Read the contents of the file
+				fileContent, err := os.ReadFile(file)
+				if err != nil {
+					fmt.Println(err)
+					errchan <- err
+					return elapsed, err
+				}
+				// Check for badness with CheckConsole
+				warnOnly, badlines := kola.CheckConsole([]byte(fileContent), nil)
+				if len(badlines) == 0 {
+					fmt.Printf("No badness detected in the file: %v\n", fileName)
+				} else {
+					for _, badline := range badlines {
+						err = fmt.Errorf("badness detected: \n%v", badline)
+						fmt.Printf("badness detected\n")
+						if err != nil {
+							errchan <- err
+							return elapsed, err
+						}
+					}
+					fmt.Printf("WarnOnly: %v\n", warnOnly)
+					if warnOnly {
+						err = fmt.Errorf("warnings found in console file")
+						errchan <- err
+						return elapsed, err
+					}
+				}
 			}
 		}
-	}()
+		return elapsed, err
```

Comment:
don't need to return here.. the return a few lines below this at the  bottom of the function should be updated to `return elapsed, err` and that is what would be used.

Reply:
Implemented in the next push.

Reply:
did you remove all of the `errchan` stuff too?

Reply:
Yes. [updated commit](https://github.com/coreos/coreos-assembler/commit/4d50394dd433f673254938162ac1a535d3026211)